enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,417 @@
+---
+title: Allowing version independence in siteconfig
+authors:
+  - "@sabbir-47"
+reviewers:
+  - "@imiller0"
+  - "@serngawy"
+  - "@Iack"
+  - "@serngawy" 
+approvers:
+  - "@imiller0"
+
+api-approvers:
+  - TBD
+creation-date: 2023-06-06
+last-updated: 2023-06-09
+status: provisional
+---
+
+
+# Cluster Group Upgrade (CGU) conditions and their reason/messages enhancement
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Documentation is updated in the upstream
+- [ ] Documentation is updated in the downstream
+
+
+## Summary
+
+Siteconfig allows customer to include custom extra manifests and list them in the siteconfig CR to be added during CR generation. It provides the flexibility to add additional CRs from the customer point of view during install time along with the CRs the RAN team provides. 
+
+The enhancement seeks to address the version independence of extra manifests, by putting version dependent manifests in version specific folder in git and pointing the directories via `extraManifestSearchPaths`.
+
+
+## Motivation
+
+* Currently customer can put custom manifests in the git repo and point them via **extraManifestPath**. 
+* **Current behavior:** If customer only want to include custom CRs from git directory to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs. But it limits:
+
+    * does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+    * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+
+    * as a result same named CR coming from user directory doesn't take precedent, rather siteconfig issues an error complaining same named CR is not allowed.
+
+  * below is an example of how to include a CR called `myCR.yaml` from `extraManifestPath`  
+
+
+  ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+
+### Goals
+
+* Customer can place the full set of version specific day-0 manifests (extra manifests) into git alongside the SiteConfig. This allows the user to deploy the correct version of manifests based on the version of OCP being deployed.
+
+* Allow customer to include multiple directory path to look for extra-manifests in siteconfig, this way we are better prepared for 4.15+ features
+
+* This will enable the customer to control what CRs get added in site generation
+
+* Keep backward compatibility of the behavior corresponds to 4.13 version, meaning the new parameter is not mandatory to generate CRs through siteconfig.
```

Comment:
For all time or for a migration period ?

Reply:
We would like to keep the behavior of 4.13 in 4.14, with adding the extra var `extraManifestSearchPaths`, if that opted by user, builtIn CRs are disabled.

So in short, we support 4.13 behavior for migration period and will deprecate from 4.15 release.